superset/common/query_context.py

```diff
@@ -115,17 +106,18 @@ def get_query_result(self, query_object: QueryObject) -> Dict[str, Any]:
             "df": df,
         }
 
+    @staticmethod
     def df_metrics_to_num(  # pylint: disable=invalid-name,no-self-use
-        self, df: pd.DataFrame, query_object: QueryObject
+        df: pd.DataFrame, query_object: QueryObject
     ) -> None:
         """Converting metrics to numeric when pandas.read_sql cannot"""
-        metrics = [metric for metric in query_object.metrics]
         for col, dtype in df.dtypes.items():
-            if dtype.type == np.object_ and col in metrics:
+            if dtype.type == np.object_ and col in query_object.metrics:
                 df[col] = pd.to_numeric(df[col], errors="coerce")
 
+    @staticmethod
```

Comment:
noice

Reply:
Thanks for setting me straight on the `ClassVar`! Will update accordingly.